src/OrchardCore.Modules/OrchardCore.Users.TimeZone/Services/IUserTimeZoneService.cs

```diff
@@ -0,0 +1,13 @@
+using System.Threading.Tasks;
+using OrchardCore.Modules;
+using OrchardCore.Users.TimeZone.Models;
+
+namespace OrchardCore.Users.TimeZone.Services
+{
+    public interface IUserTimeZoneService
+    {
+        Task<ITimeZone> GetUserTimeZoneAsync();
+        Task UpdateUserTimeZoneAsync(UserProfile profile);
```

Comment:
`GetUserTimeZoneAsync()` always calls `GetCurrentUserTimeZoneIdAsync()` which doesn't require any params to be passed to it since it only needs to know which user is currently logged in by using the `HttpContextAccessor`. On `UpdateUserTimeZoneAsync()` I passed the entire object as for example we could need to update more than one UserProfile field through this method.

Reply:
I see - the `UpdateUserTimeZoneAsync` actually updates the profile of the currently loggedin user with the profile being passed in as a prameter. Makes sense.